src/cli/cli.go

```diff
@@ -64,15 +64,24 @@ func preRun(cc *cobra.Command, args []string) error {
 	avoidTheseDescription := []string{
 		"Initialize a repository.",
 		"Initialize a S3 repository",
+		"Connect to a S3 repository",
```

Comment:
Explanation:
* Consider a case where we do a `repo connect` but the config file is not present. In that case, `preRun` will fail with this PR because `GetStorageProviderAndOverrides` will return an error(config file not found).
* We could special case this scenario for connect and source provider from the command, but it's hacky and will make it worse.
* Other alternative is to skip telemetry for connect, like we do for init. Telemetry for following backup/restore/* cmds will work as expected.

Reply:
Ah, check. Thanks for the clarity.